cmd/xl-v1-bucket.go

```diff
@@ -263,5 +277,57 @@ func (xl xlObjects) DeleteBucket(bucket string) error {
 	if errors.Cause(err) == errXLWriteQuorum {
 		xl.undoDeleteBucket(bucket)
 	}
-	return toObjectErr(err, bucket)
+	if err != nil {
+		return toObjectErr(err, bucket)
+	}
+
+	return nil
+}
+
+// SetBucketPolicies sets policy on bucket
+func (xl xlObjects) SetBucketPolicies(bucket string, policy policy.BucketAccessPolicy) error {
+	// Parse validate and save bucket policy.
+	policyBytes, err := json.Marshal(policy)
+	if err != nil {
+		return err
+	}
+	// Acquire a write lock on bucket before modifying its configuration.
+	bucketLock := xl.nsMutex.NewNSLock(bucket, "")
+	if err := bucketLock.GetLock(globalOperationTimeout); err != nil {
+		return err
+	}
+	// Release lock after notifying peers
+	defer bucketLock.Unlock()
+
```

Comment:
You can refer to the comments for fsObjects.SetBucketPolicies

Reply:
done.